src/service/resource.js

```diff
@@ -932,30 +939,45 @@ export class Resource {
     dev().fine(TAG, 'start layout:', this.debugid, 'count:', this.layoutCount_);
     this.layoutCount_++;
     this.state_ = ResourceState.LAYOUT_SCHEDULED;
+    this.abortController_ = new AbortController();
+    const abortSignal = this.abortController_.signal;
 
     const promise = new Promise((resolve, reject) => {
       Services.vsyncFor(this.hostWin).mutate(() => {
+        if (abortSignal.aborted) {
+          // The chained layoutCount_ will log the expected race error.
```

Comment:
I think we should still call `resolve()` or `reject()` here. Otherwise the outer `new Promise()` will never yield, right?